templates/bot/js/command-and-response/README.md

```diff
@@ -49,148 +47,144 @@ The following files provide the business logic for command and response bot. The
+## Extend the command bot template with more commands and responses
 
-This section outlines some customization you can do to adopt the application for your needs.
+Follow the steps below to add more commands and responses to extend the command bot:
 
-## Customize the trigger pattern
+1. [Step 1: Add a command definition in manifest](#Step-1-Add-a-command-definition-in-manifest)
+1. [Step 2: Respond with an Adaptive Card](#Step-2-Respond-with-an-Adaptive-Card)
+1. [Step 3: Handle the command](#Step-3-Handle-the-command)
+1. [Step 4: Register the new command](#Step-4-Register-the-new-command)
 
-The default pattern to trigger a command is through a defined keyword. But often times you would want to collect and process additional information retrieved after the trigger keyword. In addition to keyword match, you could also define your trigger pattern with [regular expressions](https://regex101.com/) and match against `message.text` with more controls.
+### Step 1: Add a command definition in manifest
 
-When using regular expressions, any capture group can be found in `message.matches`. Below is an example that uses regular expression to capture strings after `reboot`, for example if user inputs `reboot myMachine`, `message.matches[1]` will capture `myMachine`:
+You can edit the manifest template file `templates\appPackage\manifest.template.json` to include definitions of a command with its title and description in the `commands` array:
 
-```javascript
-class HelloWorldCommandHandler {
-  triggerPatterns = /^reboot (.*?)$/i; //"helloWorld";
-  async handleCommandReceived(context, message) {
-    console.log(`Bot received message: ${message.text}`);
-    const machineName = message.matches[1];
-    console.log(machineName);
-    // Render your adaptive card for reply message
-    const cardData = {
-      title: "Your Hello World Bot is Running",
-      body: "Congratulations! Your hello world bot is running. Click the button below to trigger an action.",
-    };
-    const cardJson = AdaptiveCards.declare(helloWorldCard).render(cardData);
-    return MessageFactory.attachment(CardFactory.adaptiveCard(cardJson));
+```json
+"commandLists": [
+  {
+    "commands": [
+        {
+            "title": "helloWorld",
+            "description": "A helloworld command to send a welcome message"
+        },
+        {
+            "title": "doSomething",
+            "description": "A sample do something command"
+        }
+    ]
   }
-}
+]
+```
 
-module.exports = {
-  HelloWorldCommandHandler,
+### Step 2: Respond with an Adaptive Card
+
+To respond with an Adaptive Card, define your card in its JSON format. Create a new file `src/adaptiveCards/doSomethingCommandResponse.json`:
+
+```json
+{
+    "type": "AdaptiveCard",
+    "body": [
+    {
+        "type": "TextBlock",
+        "size": "Medium",
+        "weight": "Bolder",
+        "text": "Your DoSomething Command is added!"
```

Comment:
`DoSomething` -> `doSomething`, better to use the same case as it is defined in the manifest.

Reply:
Done